de.fu_berlin.inf.dpp/src/de/fu_berlin/inf/dpp/ui/browser/EclipseBrowserDialog.java

```diff
@@ -46,4 +47,29 @@ private void centerShellRelativeToParent() {
             + (parentShellBounds.width - shellSize.x) / 2, parentShellBounds.y
             + (parentShellBounds.height - shellSize.y) / 2);
     }
+
+    /**
+     * If the shell is bigger than the parent, the shell size is set to 80% of
+     * the parent size.
+     */
+    private void adjustShellSize() {
+        final Composite composite = shell.getParent();
+
+        if (!(composite instanceof Shell))
+            return;
+
+        final Shell parent = (Shell) composite;
+
+        final Rectangle parentShellBounds = parent.getBounds();
+        final Rectangle shellBounds = shell.getBounds();
+
+        if (shellBounds.height > parentShellBounds.height
+            || shellBounds.width > parentShellBounds.width) {
+            int nWidth = (int) (parentShellBounds.width * 0.8);
+            int nHeight = (int) (parentShellBounds.height * 0.8);
+
+            shell.setSize(nWidth, nHeight);
```

Comment:
As a general question: Does the aspect ratio of the "child" shell matter?
Or with other words, does it matter that this adjustment does not preserver the original ration between width and height of the "child" shell?

Reply:
The shell Size was never set before, so my fix is only to prevent that the dialog is bigger that the eclipse window. I had some issues on my pc where I wasn't able to close or move the dialog because the dialog opens outside the visible area of the screen.

In general the aspect ratio does't matter for the HTML-GUI because it is responsive. Maybe it makes sense to define some ratios/sizes for different dialogs when the HTML-GUI is nearly finished. But for now it doesn't matter.

Reply:
See my comment above in the CTOR, create the shell by yourself.